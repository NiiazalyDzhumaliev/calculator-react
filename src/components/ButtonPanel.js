import Button from './Button';

const { v4: UuidV4 } = require('uuid');

const buttonGroups = {
  groupOne: {
    id: 1,
    group: ['AC', '+/-', '%', 'รท'],
  },

  groupTwo: {
    id: 2,
    group: ['7', '8', '9', 'X'],
  },

  groupThree: {
    id: 3,
    group: ['4', '5', '6', '-'],
  },

  groupFour: {
    id: 4,
    group: ['1', '2', '3', '+'],
  },

  groupFive: {
    id: 5,
    group: ['0', '.', '='],
  },
};

const ButtonPanel = () => (
  <>
    <div className="groupOne">
      {buttonGroups.groupOne.group.map(button => (
        <Button name={button} key={UuidV4()} />
      ))}
    </div>

    <div className="groupTwo">
      {buttonGroups.groupTwo.group.map(button => (
        <Button name={button} key={UuidV4()} />
      ))}
    </div>

    <div className="groupThree">
      {buttonGroups.groupThree.group.map(button => (
        <Button name={button} key={UuidV4()} />
      ))}
    </div>

    <div className="groupFour">
      {buttonGroups.groupFour.group.map(button => (
        <Button name={button} key={UuidV4()} />
      ))}
    </div>

    <div className="groupFive">
      {buttonGroups.groupFive.group.map(button => (
        <Button name={button} key={UuidV4()} />
      ))}
    </div>
  </>
);

export default ButtonPanel;
