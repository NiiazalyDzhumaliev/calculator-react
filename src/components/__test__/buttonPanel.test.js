import renderer from 'react-test-renderer';
import ButtonPanel from '../ButtonPanel';

const handleClick = buttonName => buttonName;

it('renders correctly', () => {
  const tree = renderer
    .create(<ButtonPanel clickHandler={handleClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
