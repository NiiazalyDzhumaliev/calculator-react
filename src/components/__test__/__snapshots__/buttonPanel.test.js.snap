// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="groupOne button-group"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e8813c",
          "width": "25%",
        }
      }
      type="button"
    >
      /
    </button>
  </div>,
  <div
    className="groupTwo button-group"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e8813c",
          "width": "25%",
        }
      }
      type="button"
    >
      X
    </button>
  </div>,
  <div
    className="groupThree button-group"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e8813c",
          "width": "25%",
        }
      }
      type="button"
    >
      -
    </button>
  </div>,
  <div
    className="groupFour button-group"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e8813c",
          "width": "25%",
        }
      }
      type="button"
    >
      +
    </button>
  </div>,
  <div
    className="groupFive button-group"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "50%",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e8813c",
          "width": "25%",
        }
      }
      type="button"
    >
      =
    </button>
  </div>,
]
`;
